sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Metrics/ExecutableLinesWalker.cs

```diff
@@ -81,11 +92,48 @@ private void FindExecutableLines(SyntaxNode node)
 
                 case SyntaxKind.ArrayInitializerExpression:
                     executableLineNumbers.Add(node.GetLocation().GetLineNumberToReport());
-                    return;
+                    return true;
+
+                case SyntaxKind.StructDeclaration:
+                case SyntaxKind.ClassDeclaration:
+                    var typeDeclarationSyntax = (BaseTypeDeclarationSyntax)node;
+                    return !HasExcludedCodeAttribute(typeDeclarationSyntax.AttributeLists);
+
+                case SyntaxKind.MethodDeclaration:
+                    var methodSyntax = (BaseMethodDeclarationSyntax)node;
+                    return !HasExcludedCodeAttribute(methodSyntax.AttributeLists);
+
+                case SyntaxKind.PropertyDeclaration:
+                    var propertySyntax = (PropertyDeclarationSyntax)node;
+                    return !HasExcludedCodeAttribute(propertySyntax.AttributeLists);
+
+                case SyntaxKind.EventDeclaration:
+                    var eventSyntax = (EventDeclarationSyntax)node;
+                    return !HasExcludedCodeAttribute(eventSyntax.AttributeLists);
 
                 default:
-                    return;
+                    return true;
             }
         }
+
+        private bool HasExcludedCodeAttribute(SyntaxList<AttributeListSyntax> attributeSyntaxLists)
+        {
+            var attributes = attributeSyntaxLists.SelectMany(attributeList => attributeList.Attributes);

```

Comment:
The debugging variable is not needed, you can merge this with the rest of the linq statements.